From 6af051e10fa9f2baca0037e8d5488f69555eed60 Mon Sep 17 00:00:00 2001
From: Alexander Martinz <amartinz@shiftphones.com>
Date: Thu, 28 Feb 2019 11:35:53 +0100
Subject: [PATCH 20/20] MusicPlaybackService: catch IllegalStateException for
 duration and position

On some where short tracks, which eg last one second, this can crash
the service.

Change-Id: I3db9bb70bd952554b44151b457963507174208bf
Signed-off-by: Alexander Martinz <amartinz@shiftphones.com>
---
 src/org/lineageos/eleven/MusicPlaybackService.java | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/src/org/lineageos/eleven/MusicPlaybackService.java b/src/org/lineageos/eleven/MusicPlaybackService.java
index 1725234..3283f0a 100644
--- a/src/org/lineageos/eleven/MusicPlaybackService.java
+++ b/src/org/lineageos/eleven/MusicPlaybackService.java
@@ -3396,7 +3396,12 @@ public class MusicPlaybackService extends Service {
          * @return The duration in milliseconds
          */
         public long duration() {
-            return mCurrentMediaPlayer.getDuration();
+            try {
+                return mCurrentMediaPlayer.getDuration();
+            } catch (IllegalStateException exc) {
+                Log.e(TAG, "Could not get duration", exc);
+            }
+            return 0L;
         }
 
         /**
@@ -3405,7 +3410,12 @@ public class MusicPlaybackService extends Service {
          * @return The current position in milliseconds
          */
         public long position() {
-            return mCurrentMediaPlayer.getCurrentPosition();
+            try {
+                return mCurrentMediaPlayer.getCurrentPosition();
+            } catch (IllegalStateException exc) {
+                Log.e(TAG, "Could not get current position", exc);
+            }
+            return 0L;
         }
 
         /**
-- 
2.17.1

